src/test/java/com/amplitude/AmplitudeTest.java

```diff
@@ -335,6 +339,27 @@ public void testWithSwallowMiddleware()
     assertEquals("middleware_user_2", sentEvent.userId);
   }
 
+  @Test
+  public void testShouldWait() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
This is for unit level in client. Can we have a test to use the mocked response? Return a mocked response with throttling user and assert shouldWait true then to false?